### ISSUE TITLE
Sending documents to a data stream fails. The `op_type` should be explicitly set to 'create'.

### ISSUE THREAD

**Issue by eyadmba** [vduseev/opensearch-logger#7]
On my handler I set the `index_name` instance variable to be the name of my data stream (e.g. `org-logs`).

Now when the `_bulk` request is sent to that data stream, Opensearch responds back with:
```json
{
  "took": 0,
  "errors": true,
  "items": [
    {
      "index": {
        "_index": "org-logs-2023.10.15",
        "_id": null,
        "status": 400,
        "error": {
          "type": "illegal_argument_exception",
          "reason": "only write ops with an op_type of create are allowed in data streams"
        }
      }
    }
  ]
}
```

And that is saying that this request body is invalid, because data streams only accept the "create" op_type when adding documents.
```javascript
POST _bulk

// This is invalid
{"index":{"_index":"org-logs-2023.10.15"}}
{"@timestamp":"2023-10-15T22:14:15+00:00","message":"my log"}

// It should be "create"
{"create":{"_index":"org-logs-2023.10.15"}}
...
```

Going through the code I found that the Opensearch client itself is what is defaulting to the "index" op_type because the 'actions' list it receives doesn't specify an op_type:
![image](https://github.com/vduseev/opensearch-logger/assets/13931997/7db5b4af-4eb3-4038-924d-140377d60e8b)

so I changed it to this and it did work:

![image](https://github.com/vduseev/opensearch-logger/assets/13931997/96810ab8-e735-4449-bb23-74ce98152d1c)

This is where the Opensearch client does that defaulting to "index" op_type
`...\site-packages\opensearchpy\helpers\actions.py`
![image](https://github.com/vduseev/opensearch-logger/assets/13931997/b847593d-5e7d-4a8f-aee2-0dfe0e4c6571)

---

I can create a PR to add this change for data streams only, but that would require some kind of flag to be set on the handler.
But I'm also thinking maybe there's something else to change with the index rotation logic because index rotation isn't needed for data streams; and that's why I'm opening this issue. I don't mind taking lead on this but I would need some guidance on what should be done first.

**Comment by vduseev** [comment]
For tracking purposes here are things left to do:

- [ ] Update README to reflect support for Data Streams.
- [ ] Create a new integration test that will verify the feature is working.

**Comment by eyadmba** [comment]
Ah I forgot to update the README. My bad. I can do that soon though.

**Comment by eyadmba** [comment]
Is the integration test a requirement for publishing a new version with those changes to PyPI? I was going to reference my fork, but I'd rather just use this package from PyPI directly if possible.

**Comment by vduseev** [comment]
@eyadmba, thank you very much for your valuable contribution! 

It makes me very happy and I appreciate it a lot. Thank you for your time and resources invested into making this library better. 

I've just published a new release with your code tagged as v1.3.0 to PyPI and GitHub.
